server/src/exporter.js

```diff
@@ -0,0 +1,255 @@
+const config = require("./config").root();
+const path = require("path");
+const fs = require("fs");
+const child_process = require("child_process");
+const rimraf = require("rimraf");
+
+// Convert to milliseconds:
+let keepTime = config.exportKeepTime * 60 * 1000;
+
+function exportPath(deviceId) {
+  if (deviceId.search(/^[a-z0-9\-]+$/i) == -1) {
+    reject(new Error("Bad deviceId"));
+  }
+  return path.join(config.exportBase, "export---" + deviceId);
+}
+
+function removeExportPath(dir) {
+  return new Promise((resolve, reject) => {
+    rimraf(dir, (error) => {
+      if (error && error.code != "ENOENT") {
+        reject(error);
+      } else {
+        resolve();
+      }
+    });
+  });
+}
+
+function launchWget(deviceId) {
+  return new Promise((resolve, reject) => {
+    let dir = exportPath(deviceId);
+    let script = path.join(__dirname, "export-shots.sh");
+    let start = Date.now();
+    let subprocess = child_process.spawn(
+      script,
+      {
+        env: {
+          BASE_DIR: dir,
+          AUTH: require("./dbschema").getTextKeys()[0],
+          DEVICE_ID: deviceId,
+          PORT: config.port,
+          CONTENT_PORT: config.contentPort,
+        }
+      }
+    );
+    subprocess.on("close", function (code) {
+      console.info("Exported to", dir, "in", (Date.now() - start) + "ms", "exit code:", code);
+    });
+    subprocess.on("error", function (e) {
+      console.warn("Error launching wget:", e);
+    });
+    resolve();
```

Comment:
Yes, this is synchronous, don't you want the promise to resolve inside subprocess.on("close", ...) instead? Should also reject inside on("error", ...)